src/metric/PPLCompoundMetric.py

```diff
@@ -0,0 +1,251 @@
+"""Perceptual Path Length (PPL) from the paper "A Style-Based Generator
+Architecture for Generative Adversarial Networks". The original
+implementation by Karras et al. at
+https://github.com/NVlabs/stylegan/blob/master/metrics/perceptual_path_length.py
+and the pytorch implementation in StyleGan2-ada:
+https://github.com/NVlabs/stylegan2-ada-pytorch/blob/main/metrics/perceptual_path_length.py
+"""
+# Code how to calculate the PPL is inspired by the above
+
+from src.metric.CompoundMetric import CompoundMetric
+from src.metric.SampleMetricManager import SampleMetricManager
+from src.core.Setupable import SetupMode
+import src.metric.CompoundMetricManager as CompoundMetricManager
+from typing import Any
+import torch
+from urllib import request
+from tqdm import tqdm
+import numpy as np
+from src.util.CudaUtil import get_default_device, to_device
+from src.util.FileJar import FileJar
+from pathlib import Path
+from src.generator.Generator import Generator
+from src.dataset.TorchImageDataset import TorchImageDataset
+import torchvision.transforms as T
+from torch.utils.data import DataLoader
+
+PPL_NAME = "PerceptualPathLength"
+VGG16_URL = (
+    "https://nvlabs-fi-cdn.nvidia.com/stylegan2-ada-pytorch/pretrained/metrics/vgg16.pt"
+)
+BATCH_SIZE = 64  #! BATCH_SIZE MUST BE EVEN! and >=2
+VGG16_SETUP_NAME = "VGG16_SETUP"
+VGG16_MODEL_NAME = "vgg16_zhang.pt"
+
+# Metric constants
+NUM_SAMPLES = 50000  # rec 100k stylegan paper, 50k in stylegan2-ada
+EPS = 1e-4  # According to stylegan paper
+
+#! BIM BAM BOOOP BUG ALERT
+# TODO getting unreasonable high values, 10x what they should be
+# its probably not vgg16 model since ive tried downloading it myself
+# and used a alternative source
```

Comment:
I like the comments and believe they should be kept, just check spelling, i.g., "a alternative source", "unreasonable high values"